components/style/stylesheets/keyframes_rule.rs

```diff
@@ -126,20 +126,16 @@ impl KeyframePercentage {
     }
 
     fn parse<'i, 't>(input: &mut Parser<'i, 't>) -> Result<KeyframePercentage, ParseError<'i>> {
-        let percentage = if input.try(|input| input.expect_ident_matching("from")).is_ok() {
-            KeyframePercentage::new(0.)
-        } else if input.try(|input| input.expect_ident_matching("to")).is_ok() {
-            KeyframePercentage::new(1.)
-        } else {
-            let percentage = input.expect_percentage()?;
-            if percentage >= 0. && percentage <= 1. {
-                KeyframePercentage::new(percentage)
-            } else {
-                return Err(input.new_custom_error(StyleParseErrorKind::UnspecifiedError));
-            }
-        };
-
-        Ok(percentage)
+    let token = input.next()?.clone();
```

Comment:
nit: indentation here looks wrong. Also, there's no need to clone the token except in the last branch, which can be:

```rust
ref token => Err(input.new_unexpected_token_error(token.clone()),
```